metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -98,13 +124,118 @@ final class ReferenceProvider(
     }
   }
 
+  def allInheritanceReferences(
+      symbolOccurrence: SymbolOccurrence,
+      doc: TextDocument,
+      filePosition: FilePosition,
+      fnIncludeSynthetics: Synthetic => Boolean,
+      failWhenReachingDependencySymbol: Boolean,
+      canSkipExactMatchCheck: Boolean = true
+  ): Seq[Location] = {
+    val parentSymbols = implementation
+      .topMethodParents(doc, symbolOccurrence.symbol)
+
+    if (failWhenReachingDependencySymbol && parentSymbols.exists(_.isRight)) {
+      Seq.empty
+    } else {
+
+      val topParentSymbols: Seq[Either[FilePosition, SymbolInformation]] = {
```

Comment:
Name might actually be missleading, maybe `mainDefinitions` ?

Reply:
ok